ios/brave-ios/Sources/Brave/WebFilters/FilterListResourceDownloader.swift

```diff
@@ -218,58 +162,37 @@ public actor FilterListResourceDownloader {
     return resourcesInfo
   }
 
-  /// Compile the engine from the given `AdblockService` `shieldsInstallPath` `URL`.
-  private func compileEngine(
-    filterListFolderURL folderURL: URL,
-    resourcesInfo: CachedAdBlockEngine.ResourcesInfo,
-    engineSource: CachedAdBlockEngine.Source,
-    isAlwaysAggressive: Bool,
-    compileContentBlockers: Bool
-  ) async {
-    let localFileURL = folderURL.appendingPathComponent("list.txt", conformingTo: .text)
-
-    let version = folderURL.lastPathComponent
-    let filterListInfo = CachedAdBlockEngine.FilterListInfo(
-      source: engineSource,
-      localFileURL: localFileURL,
-      version: version,
-      fileType: .text
-    )
-    let lazyInfo = AdBlockStats.LazyFilterListInfo(
-      filterListInfo: filterListInfo,
-      isAlwaysAggressive: isAlwaysAggressive
-    )
-
-    await AdBlockStats.shared.compile(
-      lazyInfo: lazyInfo,
-      resourcesInfo: resourcesInfo,
-      compileContentBlockers: compileContentBlockers
-    )
-  }
-
   /// Load general filter lists (shields) from the given `AdblockService` `shieldsInstallPath` `URL`.
   private func compileFilterListEngineIfNeeded(
-    fromComponentId componentId: String,
+    source: CachedAdBlockEngine.Source,
     folderURL: URL,
     isAlwaysAggressive: Bool,
     resourcesInfo: CachedAdBlockEngine.ResourcesInfo,
     compileContentBlockers: Bool
   ) async {
     let version = folderURL.lastPathComponent
-    let source = CachedAdBlockEngine.Source.filterList(componentId: componentId)
+    let filterListURL = folderURL.appendingPathComponent("list.txt", conformingTo: .text)
+
+    guard FileManager.default.fileExists(atPath: filterListURL.relativePath) else {
+      // We are loading the old component from cache. We don't want this file to be loaded.
+      // When we download the new component shortly we will update our cache.
+      // This should only trigger after an app update and eventually this check can be removed.
+      return
+    }
+
     let filterListInfo = CachedAdBlockEngine.FilterListInfo(
       source: source,
-      localFileURL: folderURL.appendingPathComponent("list.txt", conformingTo: .text),
+      localFileURL: filterListURL,
       version: version,
       fileType: .text
     )
     let lazyInfo = AdBlockStats.LazyFilterListInfo(
       filterListInfo: filterListInfo,
       isAlwaysAggressive: isAlwaysAggressive
     )
-    guard await AdBlockStats.shared.isEagerlyLoaded(source: source) else {
-      // Don't compile unless eager
-      await AdBlockStats.shared.updateIfNeeded(resourcesInfo: resourcesInfo)
+
+    //  Check if we should load these rules
+    guard await AdBlockStats.shared.isEnabled(source: source) else {
```

Comment:
I kinda wanna keep it the way it is. I like guards because it guarantees (it forces me to add) a return, then i don't need to worry if I added one especially if it's a complex body like this one. I know because I missed returns before and had bugs. I also know when looking back at my code (or at someone else code) that if the statement is satisfied, it won't get past that block without looking at the complexity of the block.